### ISSUE TITLE
[no-magic-numbers] False positive on negative numbers in enums w/ ignoreEnums flag on

### ISSUE THREAD

**Issue by 20ghrrdll** [typescript-eslint/typescript-eslint#1414]
Hello! I came across this odd behavior when trying to add the no-magic-numbers rule to a project: 
**Repro**

```JSON
{
  "rules": {
   'no-magic-numbers': 'off',
    '@typescript-eslint/no-magic-numbers': [
      'error',
      {
        'ignoreEnums': true,
      },
    ],
  }
}
```

```TS
enum KeyboardNav {
  IS_ACTIVE = 0,
  IS_NOT_ACTIVE = -1,
}
```

**Expected Result**
To pass without returning any errors

**Actual Result**
eslint returns: `error: No magic number: -1 (@typescript-eslint/no-magic-numbers)` .
 0, on the other hand, is not flagged.
**Additional Info**

Eslint exited with a status 1, no errors were thrown. 
When run with the debug flag, this is the dump of the output related to the file containing this enum:

```
  eslint:ignored-paths contains: +6ms
  eslint:ignored-paths   target   = "/...stuff.../shouldBeKeyboardNavAccessible.ts" +0ms
  eslint:ignored-paths   base     = "/...stuff.../" +0ms
  eslint:ignored-paths   relative = "/...stuff.../shouldBeKeyboardNavAccessible.ts" +0ms
  eslint:ignored-paths   result   = false +0ms
  eslint:file-enumerator Yield: shouldBeKeyboardNavAccessible.ts +6ms
  eslint:cli-engine Lint /...stuff.../shouldBeKeyboardNavAccessible.ts +6ms
  eslint:linter Linting code for /...stuff.../shouldBeKeyboardNavAccessible.ts (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /...stuff.../shouldBeKeyboardNavAccessible.ts +0ms
  eslint:linter Generating fixed text for/...stuff.../shouldBeKeyboardNavAccessible.ts (pass 1) +7ms
  eslint:source-code-fixer Applying fixes +7ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^2.2.0` |
| `@typescript-eslint/parser`        | `^2.2.0` |
| `TypeScript`                       | `~3.6.3` |
| `ESLint`                           | `~6.3.0` |
| `node`                             | `10.16.2` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
the rule didn't properly handle `UnaryExpression` in enums.

**Comment by 20ghrrdll** [comment]
Thanks for the speedy response!
